src/core/queryClient.ts

```diff
@@ -239,7 +239,7 @@ export class QueryClient {
     const refetchFilters: QueryFilters = {
       ...filters,
       active: filters.refetchActive ?? true,
-      inactive: filters.refetchInactive,
+      inactive: filters.refetchInactive ?? false,
```

Comment:
I found something slightly unrelated: `InvalidateQueryFilters` is defined as:

https://github.com/tannerlinsley/react-query/blob/3b2c4088e103e4b4ca990662c03fd6e67330b477/src/core/types.ts#L243-L246

`extends QueryFilters` means we can also pass `active` and `inactive` to it, but those inputs seem invalid, because they will actually be overridden with `refetchInactive` and `refetchInactive` here. I think a more correct type definition would be:

```diff
- export interface InvalidateQueryFilters extends QueryFilters {
+ export interface InvalidateQueryFilters extends Omit<QueryFilters, 'active' | 'inactive'> {
  refetchActive?: boolean
  refetchInactive?: boolean
}
```

what do you think?

Reply:
You're right. The new type is definitely more appropriate as the two values don't currently have any effect.

Reply:
I think what I said here is actually wrong. The `refetchFilters` we are building here are only used for the call to `refetchQueries` - for the actual invalidation, we use the `filters` passed to `invalidateQueries`.

That means the previous interface is actually correct, because you can pass `active` / `inactive`, and it will affect the invalidation, and you can pass `refetchActive` / `refetchInactive`, which will only affect the invalidation - sorry for that. It's curious that we don't have test for this behaviour of `invalidateQueries` :/

What I'm just wondering now too is this:

If I call `queryClient.invalidateQueries({ active: false })`, active queries will not be invalidated. But they will still be refetched because `refetchInactive` is not passed, and it falls back to `true`. So shouldn't this mapping rather read something like:

```diff
- active: filters.refetchActive ?? true,
+ active: filters.refetchActive ?? filters.active ?? true, 
```

?

Reply:
I see. So, we'll make it such that if active queries don't get invalidated, they don't get refetched either. Correct?

Reply:
would this be better? explicitly checking if `filters.active === false` better conveys the intention of the code.

```ts
const refetchFilters: QueryFilters = {
  ...filters,
  // if filters.refetchActive is not provided and filters.active is explicitly false,
  // we don't want to refetch active queries
  active: filters.refetchActive ?? (filters.active === false ? false : true),
  inactive: filters.refetchInactive ?? false,
}
```

Reply:
I think this is the same as my above code with the double ??:

```
active: filters.refetchActive ?? filters.active ?? true,
```

- first, we take `refetchActive` if provided.
- if not, we take `active` if provided
- if this is still `undefined`, we take `true`

> So, we'll make it such that if active queries don't get invalidated, they don't get refetched either. Correct?

correct. if you call `queryClient.invalidateQueries({ active: false })`, maybe with some more filters, then we would now currently still refetch those active queries, while with the new code, we would take `active:false` for `refetchActive` because it was supplied. with this, you could even _not_ invalidate, _but still_ refetch, with:

```
queryClient.refetchQueries({ active: false, refetchActive: true })
```

not sure if that makes any sense, but it's what the api allows :)

Reply:
yes, it achieves the same effect, but i thought making more explicit assertion might be better. i'm ok with your suggestion as well, since i'll also be adding some comment.

i'll add some test cases to cover the scenario too.

Reply:
nice, thank you 🙏

Reply:
done 😄